src/autotuner/genetic_autotuner.cc

```diff
@@ -148,15 +153,23 @@ llvm::Optional<MappingOptions> GeneticAutotuner::tune(
   });
   while (not tunerFinished) {
     std::this_thread::sleep_for(std::chrono::milliseconds(100));
-    if (killRequested_) {
+    if (sigint_) {
+      std::cerr
+          << "Autotuning will stop after the current generation has finished."
+          << std::endl;
+      tuner.stopAfterCurrentGeneration();
+      tunerThread.join();
```

Comment:
Hi @ttheodor , I am wondering what will happen in the case where: autotuning was running , sigint was caught, I want to do another autotuning. In that case, it seems like sigint_ will be still set and autotuning will not run. Is that correct? Thanks

Reply:
thanks a lot @ttheodor , sounds great. :)